server/core/src/logical_locking.cpp

```diff
@@ -367,7 +368,20 @@ namespace irods::logical_locking
             return ec;
         }
 
-        if (const int ec = rst::publish::to_catalog(_comm, _ctx); ec < 0) {
+        // TODO: why have the retry? What are the implications of elevating the privileges the first time?
```

Comment:
yes, what case does retrying the database failure exercise... a transient network error?  a locked database during the first attempt?

Reply:
Let's see... there's only one case I care about when it comes to unlocking with elevated privileges and that's the case where the database update failed for some non-database-related reason - be it permissions, tickets, or some other restriction imposed by iRODS. In this case, I just want to make sure that the object is unlocked.

Perhaps it would be better in the case of failure to just return the error to the caller and let them decide what to do. This is server-side only anyway, so the caller may decide to stale the offending replica or something like that (i.e. call `stale_target_replica_and_publish`).

I'm leaning toward removing the retry.

Reply:
seems reasonable.  we can't know the intent of the caller.

Reply:
This is more of a convenience function than anything, and is also internal-only so I will assume the user knows what they're doing with it and what they should do in failure. Removed the retry, thanks.